src/set_types.jl

```diff
@@ -0,0 +1,2 @@
+import NDTensors: leaf_parenttype
+NDTensors.leaf_parenttype(T::ITensor) = leaf_parenttype(NDTensors.datatype(T))
```

Comment:
If you overload with `NDTensors.leaf_parenttype` you don't need to `import`.

Reply:
I think you should move `import NDTensors: leaf_parenttype` to https://github.com/ITensor/ITensors.jl/blob/main/src/imports.jl and then define:
```julia
leaf_parenttype(T::ITensor) = leaf_parenttype(tensor(T))
```
i.e. without `NDTensors.`, and also in terms of `tensor`. That will be more general, for example when we move to the new storage type design.

Reply:
Since in the future, not all tensor storage will have `data`.